pkg/operator/ceph/cluster/osd/envs.go

```diff
@@ -32,6 +32,7 @@ const (
 	osdWalSizeEnvVarName      = "ROOK_OSD_WAL_SIZE"
 	osdsPerDeviceEnvVarName   = "ROOK_OSDS_PER_DEVICE"
 	osdDeviceClassEnvVarName  = "ROOK_OSD_DEVICE_CLASS"
+	osdConfigMapOverrideName  = "rook-ceph-osd-env-override"
```

Comment:
Can we make this more generic and apply it to all the daemons instead of just the OSD?

Reply:
I just requested the opposite [here](https://github.com/rook/rook/pull/9647#discussion_r791976522). :)

Reply:
@travisn why not? This might be useful in the future for other daemons too..., looks simple too.

Reply:
How about if we merge this and follow up on this discussion? It will help unblock @rzarzynski.

For the follow-up, this is more like a development-only feature. This change is needed to work with Crimson in the short term, but is there another scenario? For any scenario we are aware of, the operator is expected to add env vars on the pod spec instead of the admin customizing it, so I'm not sure it's worth doing this for other daemons.